pkg/controller/installation/validation.go

```diff
@@ -130,22 +130,37 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			}
 
 			if instance.Spec.CNI.Type == operatorv1.PluginCalico {
-				// Verify the specified encapsulation type is valid.
-				switch v4pool.Encapsulation {
-				case operatorv1.EncapsulationIPIP, operatorv1.EncapsulationIPIPCrossSubnet:
-					// IPIP currently requires BGP to be running in order to program routes.
-					if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
-						return fmt.Errorf("IPIP encapsulation requires that BGP is enabled")
+				if instance.Spec.CNI.IPAM.Type == operatorv1.IPAMPluginCalico {
+					// Verify the specified encapsulation type is valid.
+					switch v4pool.Encapsulation {
+					case operatorv1.EncapsulationIPIP, operatorv1.EncapsulationIPIPCrossSubnet:
+						// IPIP currently requires BGP to be running in order to program routes.
+						if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
+							return fmt.Errorf("IPIP encapsulation requires that BGP is enabled")
+						}
+					case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
+					case operatorv1.EncapsulationNone:
+						// Unencapsulated currently requires BGP to be running in order to program routes.
+						if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
+							return fmt.Errorf("Unencapsulated IP pools require that BGP is enabled")
+						}
+					default:
+						return fmt.Errorf("%s is invalid for ipPool.encapsulation, should be one of %s",
+							v4pool.Encapsulation, strings.Join(operatorv1.EncapsulationTypesString, ","))
 					}
-				case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
-				case operatorv1.EncapsulationNone:
-					// Unencapsulated currently requires BGP to be running in order to program routes.
-					if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
-						return fmt.Errorf("Unencapsulated IP pools require that BGP is enabled")
+				} else {
+					// Verify the specified encapsulation type is None for non Calico IPAM.
+					if v4pool.Encapsulation != operatorv1.EncapsulationNone {
```

Comment:
What do you think about these 2 validations for Calico CNI and non-Calico IPAM?
I've got a slight concern that even though I do not think there is a use case we support that there might be a config out there that has set VXLAN with HostLocal or something like that.

Reply:
The Calico code itself doesn't really work for VXLAN with HostLocal - we haven't implemented allocation of tunnel addresses (i.e., it would work for some traffic paths but not pod<->host).